File: Intersection of Two Arrays II/Intersection_of_Two_Arrays_II.py
class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        mpp = Counter(nums1)
        ans = []
        for i in nums2:
            if mpp[i] > 0:
                ans.append(i)
                mpp[i] -= 1
        return ans
        

File: Intersection of Two Arrays II/Intersection_of_Two_Arrays_II.java
class Solution {
    public int[] intersect(int[] nums1, int[] nums2) {
        Map<Integer, Integer> mpp = new HashMap<>();
        for (int i : nums1) {
            mpp.put(i, mpp.getOrDefault(i, 0) + 1);
        }
        List<Integer> ansList = new ArrayList<>();
        for (int i : nums2) {
            if (mpp.containsKey(i)) {
                ansList.add(i);
                mpp.put(i, mpp.get(i) - 1);
                if (mpp.get(i) == 0) {
                    mpp.remove(i);
                }
            }
        }
        int[] ans = new int[ansList.size()];
        for (int i = 0; i < ansList.size(); i++) {
            ans[i] = ansList.get(i);
        }
        return ans;
    }
}